status/raw/status-app-protocols.md

```diff
@@ -0,0 +1,372 @@
+or to populate the local message database when the client starts up for the first time.
+
+**Store queries for reliability**
+
+Status clients MAY use periodic content filtered queries with `include_data` set to `false`,
+to retrieve only the message hashes of past messages on content topics relevant to the client.
+This can be used to compare the hashes available in the local message database with the hashes in the query response
+in order to identify possible missing messages.
+Once the Status client has identified a set of missing message hashes
+it SHOULD use [message hash lookup queries](https://github.com/waku-org/specs/blob/8fea97c36c7bbdb8ddc284fa32aee8d00a2b4467/standards/core/store.md#message-hash-lookup-queries) with `include_data` set to `true`
+to retrieve the full contents of the missing messages based on the hash.
+
+Status clients MAY use [presence queries](https://github.com/waku-org/specs/blob/8fea97c36c7bbdb8ddc284fa32aee8d00a2b4467/standards/core/store.md#presence-queries)
+to determine if one or more message hashes known to the client is present in the store service node.
+Clients MAY use this method to determine if a message that originated from the client
+has been successfully stored.
+
+**Self-addressed messages**
+
+Status clients (full or light) SHOULD use store queries (rather than subscriptions) to retrieve self-addressed messages relevant to that client.
+See [Self-addressed messages](#self-addressed-messages).
+
+**Large messages**
+
+Status clients (full or light) SHOULD use store queries (rather than subscriptions) to retrieve large messages relevant to that client.
+See [Large messages](#large-messages).
+
+### Providing services
+
+Status clients MAY provide service-side protocols to other clients.
+
+Full clients SHOULD mount
+the filter service protocol (see [12/WAKU2-FILTER](../../waku/standards/core/12/filter.md))
+and lightpush service protocol (see [19/WAKU2-LIGHTPUSH](../../waku/standards/core/19/lightpush.md))
+in order to provide light subscription and publishing services to other clients 
+for each pubsub topic to which they have a relay subscription.
+
+Status clients MAY mount the store query protocol as service node (see [WAKU2-STORE](https://github.com/waku-org/specs/blob/8fea97c36c7bbdb8ddc284fa32aee8d00a2b4467/standards/core/store.md))
+to store historical messages and
+provide store services to other clients
+for each pubsub topic to which they have a relay subscription
+
+### Self-addressed messages
+
+Messages with a _local_ functional scope (see [Functional scope](#functional-scope)),
+also known as _self-addressed_ messages,
+MUST be published to a distinct pubsub topic or a distinct _set_ of pubsub topics
+used exclusively for messages with local scope (see [Pubsub topics and sharding](#pubsub-topics-and-sharding)).
+Status clients (full or light) MUST use lightpush protocol to publish self-addressed messages (see [Publishing](#publishing)).
```

Comment:
Shall we set expectations for the service node (eg usage of store sync over relay)?

Reply:
I think this is underdefined for now.